test/TestContext.ts

```diff
@@ -2,30 +2,44 @@ import type { Cradle } from '@fastify/awilix'
 import { NewRelicTransactionManager } from '@lokalise/fastify-extras'
 import { globalLogger } from '@lokalise/node-core'
 import type { AwilixContainer } from 'awilix'
-import { createContainer } from 'awilix'
+import { asFunction, createContainer } from 'awilix'
 import type { FastifyInstance } from 'fastify'
+import merge from 'ts-deepmerge'
 
+import { getConfig, type ConfigOverrides } from '../src/infrastructure/config'
 import type { DependencyOverrides } from '../src/infrastructure/diConfig'
 import { registerDependencies } from '../src/infrastructure/diConfig'
 
 export type TestContext = {
   diContainer: AwilixContainer<Cradle>
 }
 
-export function createTestContext(dependencyOverrides: DependencyOverrides = {}): TestContext {
+export function createTestContext(
+  dependencyOverrides: DependencyOverrides = {},
+  configOVerrides: ConfigOverrides = {},
+): TestContext {
   const diContainer = createContainer({
     injectionMode: 'PROXY',
   })
+
   const fakeApp: Partial<FastifyInstance> = {
     newrelicTransactionManager: new NewRelicTransactionManager(false),
   }
+
+  const dependencies = {
+    ...dependencyOverrides,
+    config: asFunction(() => {
```

Comment:
Keep in mind this will break tests that are currently manually adding `config` to `dependencyOverrides`.

Alternatively switch this line with `...dependencyOverrides` above.